sim/core/spell_outcome.go

```diff
@@ -496,14 +494,10 @@ func (dot *Dot) outcomeRangedHitAndCritSnapshot(sim *Simulation, result *SpellRe
 	chance := 0.0
 
 	if dot.Spell.Unit.PseudoStats.InFrontOfTarget {
-		if !result.applyAttackTableMissNoDWPenalty(dot.Spell, attackTable, roll, &chance) {
-			if result.applyAttackTableCritSeparateRollSnapshot(sim, dot) {
-				result.applyAttackTableBlock(dot.Spell, attackTable, roll, &chance)
-			} else {
-				if !result.applyAttackTableBlock(dot.Spell, attackTable, roll, &chance) {
-					result.applyAttackTableHit(dot.Spell, countHits)
-				}
-			}
+		if !result.applyAttackTableMissNoDWPenalty(dot.Spell, attackTable, roll, &chance) &&
+			!result.applyAttackTableDodge(dot.Spell, attackTable, roll, &chance) &&
+			!result.applyAttackTableCritSeparateRollSnapshot(sim, dot) {
+			result.applyAttackTableHit(dot.Spell, countHits)
 		}
 	} else {
 		if !result.applyAttackTableMissNoDWPenalty(dot.Spell, attackTable, roll, &chance) &&
```

Comment:
Same here, you're missing Dodge outcomes when attacking from behind, and shouldn't need to branch off `InFrontOfTarget` anymore in MoP if I'm understanding the changes correctly.